### PR TITLE
Furniture loss of power event

### PR DESCRIPTION
**Pull request by Raketfart**
I added a trigger for furniture, when they lose power, as per issue #1312.

I used the new animation system for:

Oxygen generator - now animates, and when power is off it returns to idle state.

Oxygen compressor - This doesn't return to an idle state when turned off, but it leaks half of it's compressed gas. 
I'm not sure if that's the effect it should have, but it's a nice testcase for this event trigger, as we don't always want furniture to return to "idle" when unpowered.

Feedback is appreciated!

### CODE REVIEW

File: Assets/StreamingAssets/LUA/Furniture.lua [hunk 1]
```diff
@@ -467,16 +473,18 @@ function OxygenCompressor_OnUpdate(furniture, deltaTime)
             furniture.UpdateOnChanged(furniture)
         end
     end
+	furniture.SetAnimationState("running")
```
Comment (TomMalbran) [inline Assets/StreamingAssets/LUA/Furniture.lua]: We should use spaces.

File: Assets/StreamingAssets/LUA/Furniture.lua [hunk 2]
```diff
@@ -31,8 +31,14 @@ function OnUpdate_GasGenerator( furniture, deltaTime )
         end
     end
 	return
+	furniture.SetAnimationState("running")
 end
 
+function OnPowerOff_GasGenerator( furniture, deltaTime )
```
Comment (TomMalbran) [inline Assets/StreamingAssets/LUA/Furniture.lua]: Is the convention to have first the furniture and then the action or the other way around. I like first Furniture then action.

Comment (Raketfart) [inline Assets/StreamingAssets/LUA/Furniture.lua]: I'm not sure... we have half one and half the other. I think I agree with you - I was just trying to be consistent

Comment (TomMalbran) [inline Assets/StreamingAssets/LUA/Furniture.lua]: We have an issue #765, where it looks like we agree on `funitureName_functionName`. But the furniture names might change in the future.

**Comment by Raketfart** [comment]
I renamed the actions to furniture_action instead of the other way around, and renamed gasgenerator to oxygengenerator

**Comment by koosemose** [comment]
Looks good, Stylecop clean. Merged.

**Pull request merged**